src/background/Wallet/model/ethers-v5-types.ts

```diff
@@ -0,0 +1,193 @@
+  blockNumber?: number;
+  blockHash?: string;
+
+  // Not optional (as it is in Transaction)
+  from: string;
+
+  // The raw transaction
+  raw?: string;
+}
+
+function fromEthersBigNumber(
+  x: EthersV5BigNumber | number | string | bigint | null | undefined
+) {
+  if (x == null) {
+    return null;
+  }
+  if (EthersV5BigNumber.isBigNumber(x)) {
+    return BigInt(EthersV5BigNumber.from(x).toHexString());
+  } else {
+    return BigInt(x);
+  }
+}
+
+export function v5ToPlainTransactionResponse(
+  value: EthersV5TransactionResponse
+): TransactionResponsePlain {
+  return {
+    blockHash: value.blockHash ?? null,
+    blockNumber: value.blockNumber ?? null,
+    chainId: valueToHex(fromEthersBigNumber(value.chainId) ?? 'never'),
+    data: value.data,
+    from: value.from,
+    gasLimit: (fromEthersBigNumber(value.gasLimit) ?? 0n).toString(),
+    gasPrice: (fromEthersBigNumber(value.gasPrice) ?? 0n).toString(),
+    hash: value.hash,
+    index: 0 /** Not sure? */,
+    maxFeePerBlobGas: null,
+    maxFeePerGas: fromEthersBigNumber(value.maxFeePerGas)?.toString() ?? null,
+    maxPriorityFeePerGas:
+      fromEthersBigNumber(value.maxPriorityFeePerGas)?.toString() ?? null,
+    nonce: Number(fromEthersBigNumber(value.nonce) ?? 0),
+    to: value.to ?? null,
+    type: value.type ?? 0,
+    value: (fromEthersBigNumber(value.value) ?? 0n).toString(),
+  };
+}
+
+export function toPlainTransactionResponse(
+  value: PartiallyOptional<ethers.TransactionResponse, 'gasPrice' | 'signature'>
+): TransactionResponsePlain {
+  return {
+    blockHash: value.blockHash,
+    blockNumber: value.blockNumber,
+    chainId: valueToHex(value.chainId),
+    data: value.data,
+    from: value.from,
+    gasLimit: value.gasLimit.toString(),
+    gasPrice: value.gasPrice?.toString() ?? null,
+    hash: value.hash,
+    index: value.index,
+    maxFeePerBlobGas: value.maxFeePerBlobGas?.toString() ?? null,
+    maxFeePerGas: value.maxFeePerGas?.toString() ?? null,
+    maxPriorityFeePerGas: value.maxPriorityFeePerGas?.toString() ?? null,
+    nonce: value.nonce,
+    to: value.to,
+    type: value.type,
+    value: value.value.toString(),
+  };
+}
+
+export function toEthersV5TransactionResponse(
+  /**
+   * Can't trust TransactionResponse type because ethers@v6 is buggy:
+   * https://github.com/zksync-sdk/zksync-ethers/issues/224
+   * https://github.com/ethers-io/ethers.js/issues/4891
+   */
+  value: Partial<ethers.TransactionResponse>
+): EthersV5TransactionResponse {
+  invariant(value.chainId, 'chainId is required');
+  invariant(value.from, 'from is required');
+  invariant(value.hash, 'hash is required');
+  invariant(value.nonce, 'nonce is required');
+  return {
+    chainId: Number(value.chainId),
+    data: value.data || '',
+    from: value.from,
+    to: value.to ?? undefined,
+    gasLimit: value.gasLimit?.toString() ?? '0',
+    hash: value.hash,
+    nonce: value.nonce,
+    value: value.value?.toString() ?? '0',
```

Comment:
Shouldn't it be 0x0 here by default? I don't remember why, but we used hex value a lot.

Reply:
Yeah I hesitated, too, but I think what matters is that this string parses to a number. And both `'0'` and `'0x0'` do

Reply:
And with since these fields are now typed as bigints, I thought that this is more representative because `BigInt(0).toString() === '0'`